specification/search/Azure.Search/models-index.tsp

```diff
@@ -0,0 +1,1072 @@
+ * of search queries.
+ */
+union ScoringStatistics {
+  string,
+
+  /**
+   * The scoring statistics will be calculated locally for lower latency.
+   */
+  Local: "local",
+
+  /**
+   * The scoring statistics will be calculated globally for more consistent scoring.
+   */
+  Global: "global",
+}
+
+/**
+ * The operation to perform on a document in an indexing batch.
+ */
+union IndexActionType {
+  string,
+
+  /**
+   * Inserts the document into the index if it is new and updates it if it exists.
+   * All fields are replaced in the update case.
+   */
+  Upload: "upload",
+
+  /**
+   * Merges the specified field values with an existing document. If the document
+   * does not exist, the merge will fail. Any field you specify in a merge will
+   * replace the existing field in the document. This also applies to collections of
+   * primitive and complex types.
+   */
+  Merge: "merge",
+
+  /**
+   * Behaves like merge if a document with the given key already exists in the
+   * index. If the document does not exist, it behaves like upload with a new
+   * document.
+   */
+  MergeOrUpload: "mergeOrUpload",
+
+  /**
+   * Removes the specified document from the index. Any field you specify in a
+   * delete operation other than the key field will be ignored. If you want to
+   * remove an individual field from a document, use merge instead and set the field
+   * explicitly to null.
+   */
+  Delete: "delete",
+}
+
+/**
+ * Specifies the mode for Autocomplete. The default is 'oneTerm'. Use 'twoTerms'
+ * to get shingles and 'oneTermWithContext' to use the current context in
+ * producing autocomplete terms.
+ */
+union AutocompleteMode {
+  string,
+
+  /**
+   * Only one term is suggested. If the query has two terms, only the last term is
+   * completed. For example, if the input is 'washington medic', the suggested terms
+   * could include 'medicaid', 'medicare', and 'medicine'.
+   */
+  OneTerm: "oneTerm",
+
+  /**
+   * Matching two-term phrases in the index will be suggested. For example, if the
+   * input is 'medic', the suggested terms could include 'medicare coverage' and 'medical assistant'.
+   */
+  TwoTerms: "twoTerms",
+
+  /**
+   * Completes the last term in a query with two or more terms, where the last two
+   * terms are a phrase that exists in the index. For example, if the input is 'washington medic', the suggested terms could include 'washington medicaid' and 'washington medical'.
+   */
+  OneTermWithContext: "oneTermWithContext",
+}
+
+/**
+ * Response containing search results from an index.
+ */
+model SearchDocumentsResult {
+  /**
+   * The total count of results found by the search operation, or null if the count
+   * was not requested. If present, the count may be greater than the number of
+   * results in this response. This can happen if you use the $top or $skip
+   * parameters, or if the query can't return all the requested documents in a
+   * single response.
+   */
+  @visibility("read")
```

Comment:
Since this seems to be an output only model you dont need to specify the visibility read decorators on the fields of this model.

Reply:
This feedback applies widely across this spec. I saw Johan recently provide similar feedback on the KV spec